compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -1273,6 +1273,9 @@ impl<'tcx> InferCtxt<'tcx> {
     where
         T: TypeFoldable<TyCtxt<'tcx>>,
     {
+        if let Err(guar) = value.error_reported() {
+            self.set_tainted_by_errors(guar);
+        }
```

Comment:
While this could be done in a more targeted location to just fix this one specific kind of ICE, as long as there is no perf impact, this seems like a good place to check a lot of types for errors